### PR TITLE
feat: add A2cTrainer and un-clipped A2C loss with smoke tests

### LINKED ISSUE #152: A2C: A2cTrainer + a2c/loss.rs + always-on smoke test
## Background

Part of the A2C epic #150 (item 6, **PR B**). With `A2cConfig` landed (#151), this issue implements the actual A2C trainer + loss math. A2C is the *single-update-per-rollout, no-clipping* special case of the actor-critic family, so it REUSES PPO's policy (`MlpBurnPolicy`), optimizer (`BurnOptimizer`), and GAE path, and only diverges in the loss and the loop.

**Depends on #151** (needs `A2cConfig`).

## A2C loss (new, `src/train/a2c/loss.rs`)

- Policy-gradient loss: `-mean(log_prob * advantage_detached)` — NO importance ratio, NO clipping (this is the key divergence from PPO's `compute_policy_loss`).
- Value loss: `0.5 * mean((V(s) - returns)^2)` — plain MSE, NO value-clip.
- Entropy loss: `-mean(entropy)` (reuse `compute_entropy_loss` from `ppo::loss`, or re-implement the one-liner).
- `total = pg_loss + value_coef * value_loss + entropy_coef * entropy_loss`.
- Reuse `scalar_f64` from `ppo::loss` for host-side stat extraction.

## A2C trainer (new, `src/train/a2c/trainer.rs`)

- `A2cTrainer<B, P, O>` with the SAME move-through `Option<P>` ownership model as `PPOTrainerBurn` (Burn's optimizer consumes the module by value).
- `new(config: A2cConfig, policy: P, optimizer: BurnOptimizer<B, P, O>) -> Result<Self>` (calls `config.validate()`; seeds an internal `StdRng` from `config.seed` if needed).
- `train_step(observations: Tensor<B,2>, actions: Tensor<B,1,Int>, advantages: Tensor<B,1>, returns: Tensor<B,1>, evaluate_fn) -> Result<A2cStats>` where `evaluate_fn: FnMut(&P, Tensor<B,2>, Tensor<B,1,Int>) -> (log_probs, entropy, values)` (same closure shape as PPO).
  - NOTE: NO `old_log_probs` / `old_values` args — A2C is on-policy with no importance ratio.
  - ONE gradient step per call: no epoch loop, no minibatch shuffle, no KL early-stop.
  - Optional advantage normalization (zero-mean/unit-var) when `config.normalize_advantages`.
  - Optional global-norm grad clip via `config.max_grad_norm`.
  - Keep the PPO entropy-collapse guard (generic).
- Accessors mirroring PPO: `config()`, `policy()`, `total_steps()`, `total_episodes()`, `increment_episodes(n)`.
- `A2cStats` may reuse `ppo::TrainingStats` or be a small purpose-built struct with `policy_loss`, `value_loss`, `entropy`, `total_loss` (all `f64`, finite).

## Always-on fast smoke test (item 3 of the epic)

In `trainer.rs` `#[cfg(test)]` on `Autodiff<NdArray<f32>>`:
- `a2c_trainer_constructs` — builds with `MlpBurnPolicy` + `BurnOptimizer`, `total_steps() == 0`.
- `a2c_train_step_runs` — one `train_step` against a tiny synthetic batch (mirror the `ppo_trainer_burn_train_step_runs` fixture), asserting `total_steps() == 1` and all `A2cStats` fields finite.

## Acceptance criteria

- [ ] `src/train/a2c/loss.rs` with the unclipped PG loss, plain-MSE value loss, entropy loss, plus unit tests (loss shapes `[1]`, PG loss sign sanity, value loss equals plain MSE).
- [ ] `src/train/a2c/trainer.rs` with `A2cTrainer` per the surface above.
- [ ] The two always-on smoke tests pass.
- [ ] `mod.rs` re-exports `A2cTrainer` (+ `A2cStats`); `train/mod.rs` re-export updated.
- [ ] `cargo build --features training` and `cargo test --features training` pass; `cargo clippy --features training` clean.

## Affected files

- NEW `src/train/a2c/loss.rs`
- NEW `src/train/a2c/trainer.rs`
- `src/train/a2c/mod.rs`, `src/train/mod.rs` (re-exports)

## References

- `src/train/ppo/trainer.rs` (ownership model, `train_step` shape, advantage normalization, entropy-collapse guard), `src/train/ppo/loss.rs` (`compute_entropy_loss`, `scalar_f64`), `src/train/optimizer.rs`, `src/policy/mlp.rs` (`evaluate_actions`).

Depends on #151. Part of #150.

### PR DESCRIPTION
**Pull request by rjwalters**
## Summary

Implements PR B of the A2C epic (#150): the `A2cTrainer` and the A2C loss math, landing as a sibling to the PPO Burn trainer. With `A2cConfig` already merged (#151), this PR adds the trainer + loss and fast always-on smoke tests. It reuses PPO infrastructure (`MlpBurnPolicy`, `BurnOptimizer`, the `Option<P>` move-through ownership model, host-side advantage normalization) and only diverges in the loss and the loop.

## Changes

- **NEW `src/train/a2c/loss.rs`**:
  - `compute_a2c_policy_loss(log_probs, advantages) = -mean(log_prob * advantage)` — un-clipped, ratio-free policy gradient. Advantages are detached (constants w.r.t. params).
  - `compute_a2c_value_loss(values, returns) = mean((V(s) - returns)^2)` — plain MSE, no value-clip, no pessimistic max. Returns detached.
  - Re-exports PPO's `compute_entropy_loss` (`-mean(entropy)`) and `scalar_f64` rather than re-implementing.
- **NEW `src/train/a2c/trainer.rs`**:
  - `A2cTrainer<B, P, O>` with the same `Option<P>` move-through ownership as `PPOTrainerBurn`.
  - `new(config, policy, optimizer)` calls `config.validate()`, seeds an internal `StdRng` from `config.seed`, and stages `max_grad_norm` on the optimizer wrapper.
  - `train_step(observations, actions, advantages, returns, evaluate_fn)` — note **no** `old_log_probs`/`old_values`. Optional advantage normalization (`normalize_advantages`), one gradient step per rollout (no epochs, no shuffle, no KL early-stop). Total loss = `policy_loss + value_coef*value_loss + entropy_coef*entropy_loss`. PPO entropy-collapse guard retained.
  - Accessors `config()`, `policy()`, `total_steps()`, `total_episodes()`, `increment_episodes(n)`.
  - Purpose-built `A2cStats { policy_loss, value_loss, entropy, total_loss }` (no clip-fraction/KL/explained-var — A2C has no such diagnostics).
- **Re-exports** wired through `src/train/a2c/mod.rs` and `src/train/mod.rs`.

## Loss math

```
policy_loss  = -mean(log_prob * advantage)            # no ratio, no clip
value_loss   =  mean((V(s) - returns)^2)              # plain MSE, no clip
entropy_loss = -mean(entropy)
total        = policy_loss + value_coef*value_loss + entropy_coef*entropy_loss
```

## Acceptance Criteria Verification

| Criterion | Status | Verification |
|-----------|--------|--------------|
| `a2c/loss.rs` with unclipped PG loss, plain-MSE value loss, entropy loss + unit tests (shapes `[1]`, PG sign, value=MSE) | ✅ | 5 unit tests pass incl. shape `[1]`, positive-advantage sign, MSE-equals check |
| `a2c/trainer.rs` with `A2cTrainer` per the surface above | ✅ | Full surface implemented; `train_step` drops `old_*` args |
| Two always-on smoke tests pass | ✅ | `a2c_trainer_constructs`, `a2c_train_step_runs` (asserts `total_steps()==1`, all stats finite) |
| `mod.rs` re-exports `A2cTrainer` (+ `A2cStats`); `train/mod.rs` updated | ✅ | Both updated |
| `cargo build --features training` passes | ✅ | Clean build |
| `cargo test --features training` passes | ✅ | 10 a2c lib tests pass (7 new + 3 config) |
| `cargo clippy --features training` clean | ✅ | Zero clippy issues in touched files; pre-existing errors live only in untouched files (snake env, buffer/rollout/sampling.rs) and are out of scope |

## Test Plan

- `cargo build --features training` — clean.
- `cargo test --features training --lib a2c` — 10 passed, 0 failed.
- `cargo fmt --check` — clean.
- `cargo clippy --all-targets --features training -- -D warnings` — no issues in `src/train/a2c/*`; remaining errors are pre-existing in untouched files (verified the `sampling.rs` warning exists at base commit).
- `RUSTDOCFLAGS="-D warnings" cargo doc --no-deps --all-features` — exit 0, no doc warnings.

Additive only — the PPO trainer is untouched. The slow CartPole convergence test is intentionally out of scope (that is #153).

Closes #152

### CODE REVIEW

**Comment by rjwalters** [comment]
## Judge Review: APPROVED

Reviewed with extra scrutiny on the A2C loss math (`src/train/a2c/loss.rs`) and trainer loop. Everything checks out.

### Loss correctness (the error-prone part) — all correct
- **Policy loss** (`compute_a2c_policy_loss`): `-mean(log_prob * advantage)`. No importance ratio, no clipping. Advantages **detached** (`advantages.detach()`) so no gradient flows through them. ✅
- **Value loss** (`compute_a2c_value_loss`): `mean((V(s) - returns)^2)` — plain MSE, no value-clip, no pessimistic max. Returns **detached**. ✅
- **Entropy + total**: reuses PPO's `compute_entropy_loss = -mean(entropy)` verbatim; `total = policy_loss + value_coef*value_loss + entropy_coef*entropy_loss`. Sign convention is **identical** to PPO trainer (negative-mean-entropy added with positive coef → minimizing total maximizes entropy). ✅

### Trainer divergence from PPO
- Exactly one gradient step per rollout: no epoch loop, no minibatch shuffle, no KL early-stop. ✅
- `train_step` correctly drops `old_log_probs`/`old_values`. ✅
- Gradient flow correct: `evaluate_fn(&policy, ...)` keeps log_probs/values grad-bearing; advantages/returns detached → policy-gradient with detached advantages. ✅

### Config knobs
- `max_grad_norm` staged via `clip_grad_norm` (matches PPO/SAC wrapper pattern; real clip in inner `AdamConfig::with_grad_clipping`). `normalize_advantages`, `value_coef`/`entropy_coef`, and `seed` all honored. ✅

### Additive only
`git diff --stat main -- src/train/ppo/` is empty — PPO trainer untouched. Only a2c module + two re-export lines. ✅

### Smoke tests
`a2c_trainer_constructs` and `a2c_train_step_runs` (asserts `total_steps()==1`, all stats finite). Run in 0.01s — fast and always-on. Slow convergence test correctly deferred to #153. ✅

### Local verification
- `cargo test --features training --lib a2c` → 10 passed, 0 failed
- `cargo fmt --check` → clean
- `RUSTDOCFLAGS="-D warnings" cargo doc --no-deps --all-features` → exit 0
- `cargo clippy --features training --lib` → zero issues in touched a2c files (remaining lib warnings are pre-existing in untouched files)

Full Tests CI is the merge-queue gate (~1-2h) and not blocked on here per policy. Fast checks green; `gh pr checks` shows Rustfmt + Check(ubuntu) passing, others pending.

LGTM.
